dinov2/train/train.py

```diff
@@ -205,45 +251,91 @@ def do_train(cfg, model, resume=False):
 
     dataset = make_dataset(
         dataset_str=cfg.train.dataset_path,
-        transform=data_transform,
+        transform=data_transform_cpu,
         target_transform=lambda _: (),
+        with_targets=False,
+        cache_dataset=cfg.train.cache_dataset,
     )
     # sampler_type = SamplerType.INFINITE
     sampler_type = SamplerType.SHARDED_INFINITE
-    data_loader = make_data_loader(
-        dataset=dataset,
-        batch_size=cfg.train.batch_size_per_gpu,
-        num_workers=cfg.train.num_workers,
-        shuffle=True,
-        seed=start_iter,  # TODO: Fix this -- cfg.train.seed
-        sampler_type=sampler_type,
-        sampler_advance=0,  # TODO(qas): fix this -- start_iter * cfg.train.batch_size_per_gpu,
-        drop_last=True,
-        collate_fn=collate_fn,
-    )
+    if cfg.train.augmentations == AugmentationType.TORCHV_CPU.value:
+        data_loader = make_data_loader(
+            dataset=dataset,
+            batch_size=cfg.train.batch_size_per_gpu,
+            num_workers=cfg.train.num_workers,
+            shuffle=True,
+            seed=start_iter,  # TODO: Fix this -- cfg.train.seed
+            sampler_type=sampler_type,
+            sampler_advance=0,  # TODO(qas): fix this -- start_iter * cfg.train.batch_size_per_gpu,
+            drop_last=True,
+            collate_fn=collate_fn,
+        )
+    else:
+        data_loader = make_data_loader(
+            dataset=dataset,
+            batch_size=cfg.train.batch_size_per_gpu,
+            num_workers=cfg.train.num_workers,
+            shuffle=True,
+            seed=start_iter,  # TODO: Fix this -- cfg.train.seed
+            sampler_type=sampler_type,
+            sampler_advance=0,  # TODO(qas): fix this -- start_iter * cfg.train.batch_size_per_gpu,
+            drop_last=True,
+            collate_fn=None,
+        )
 
     # training loop
 
     iteration = start_iter
+    tot_nb_seen_samples = 0
 
     logger.info("Starting training from iteration {}".format(start_iter))
     metrics_file = os.path.join(cfg.train.output_dir, "training_metrics.json")
     metric_logger = MetricLogger(delimiter="  ", output_file=metrics_file, verbose=distributed.is_main_process())
     header = "Training"
 
+    if cfg.train.do_profiling:
+        profiler_schedule = torch.profiler.profiler.schedule(wait=0, warmup=1, active=1, repeat=0, skip_first=0)
+
+        print("------- STARTING PROFILER -------")
+        activities = [ProfilerActivity.CPU, ProfilerActivity.CUDA]
+        profiler_dir = os.path.join(cfg.train.output_dir, "profiler")
+        os.makedirs(profiler_dir, exist_ok=True)
+        profiler = torch.profiler.profile(
+            schedule=profiler_schedule,
+            activities=activities,
+            on_trace_ready=torch.profiler.tensorboard_trace_handler(profiler_dir),
+            with_stack=True,
+        )
+        profiler.start()
```

Comment:
does profiling work now? Otherwise, maybe we should not include it in main yet?

Reply:
I think it should be included because I am confident that the code is correct (copied from tutorial), and this code should work once we upgrade to PyTorch 2.1